application/src/main/java/org/thingsboard/server/controller/DeviceController.java

```diff
@@ -291,6 +290,7 @@ public DeviceCredentials getDeviceCredentialsByDeviceId(@PathVariable(DEVICE_ID)
     public DeviceCredentials saveDeviceCredentials(@RequestBody DeviceCredentials deviceCredentials) throws ThingsboardException {
         checkNotNull(deviceCredentials);
         try {
+            deviceCredentialsService.verifySecurityKeyDevice(deviceCredentials);
```

Comment:
Wrong place. The validation should happen in the deviceCredentialsService.updateDeviceCredentials. See  credentialsValidator inside DeviceCredentialsServiceImpl for this purpose.